Better error messages (show where error is)

Hey guys, when renaming some of my components I get the following error. Which is a good thing. However, the stack trace or error message never tells me *where* it is complaining. Would be nice to have this show, would save me time figuring out where it is in my project.
```
  Error: Render async fragment error (lasso-slot:head). Exception: Error: File to import not found or unreadable: application/client/components/s__navigation/include.
      at module.exports.render.options.error (/home/karl/dev/langurama/node_modules/node-sass/lib/index.js:291:26)
  Creation stack trace: Error
      at AsyncStream.beginAsync (/home/karl/dev/langurama/node_modules/marko/src/runtime/html/AsyncStream.js:192:56)
      at render (/home/karl/dev/langurama/node_modules/@lasso/marko-taglib/taglib/slot-tag.js:128:24)
      at render (/home/karl/dev/langurama/node_modules/@lasso/marko-taglib/taglib/head-tag.js:16:3)
      at wrappedRenderer (/home/karl/dev/langurama/node_modules/marko/src/runtime/helpers.js:278:13)
      at render (/home/karl/dev/langurama/application/client/pages/home/home.marko.js:36:3)
      at Template.renderer (/home/karl/dev/langurama/node_modules/marko/src/components/renderer.js:217:9)
      at hotReloadProxy (/home/karl/dev/langurama/node_modules/marko/src/hot-reload.js:101:31)
      at safeRender (/home/karl/dev/langurama/node_modules/marko/src/runtime/renderable.js:6:9)
      at Template.render (/home/karl/dev/langurama/node_modules/marko/src/runtime/renderable.js:136:20)
      at Readable._read (/home/karl/dev/langurama/node_modules/marko/src/runtime/html/Template.js:37:18)
      at AsyncStream.error (/home/karl/dev/langurama/node_modules/marko/src/runtime/html/AsyncStream.js:441:13)
      at module.exports.lassoPageResultAsyncValue.done.process.nextTick (/home/karl/dev/langurama/node_modules/@lasso/marko-taglib/taglib/slot-tag.js:139:20)
      at process._tickCallback (internal/process/next_tick.js:61:11)
```

To me, the ideal is [Rust’s error messages](https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/improved-error-messages.html). Their design philosophy ([Rust blog: _Shape of errors to come_](https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)) makes a ton of sense, and I would be delighted if Marko completely copied how they do it:

> ```
> error[E0506]: cannot assign to `x` because it is borrowed
>  --> foo.rs:6:5
>   |
> 4 |     let y = &x;
>   |              - borrow of `x` occurs here
> 5 |
> 6 |     x += 1;
>   |     ^^^^^^ assignment to borrowed `x` occurs here
> 
> error: aborting due to previous error
> ```
> 
> This error isn't terribly different, but shows off how the format has changed. It shows off your code in context, rather than just showing the text of the lines themselves.

--- a/src/core-tags/core/await/renderer.js
+++ b/src/core-tags/core/await/renderer.js
@@ -50,10 +50,19 @@ function requestData(provider, timeout) {
         timeout = 10000;
     }
 
+    var error;
+    var errorMsg = "Timed out after " + timeout + "ms";
+    // eslint-disable-next-line no-constant-condition
+    if ("MARKO_DEBUG") {
+        // Make sure we have a meaningful stack trace in development preparing the stacktrace upfront.
+        // If we create it inside the setTimeout, we will end up with a short, not meaningful, stack trace
+        // We only do it in development to avoid overhead in production
+        error = new Error(errorMsg);
+    }
     if (timeout > 0) {
         let timeoutId = setTimeout(function() {
             timeoutId = null;
-            var error = new Error("Timed out after " + timeout + "ms");
+            if (!error) error = new Error(errorMsg);
             error.code = "ERR_AWAIT_TIMEDOUT";
             error.name = "TimeoutError";
             asyncValue.___reject(error);

